x-pack/plugins/security_solution/public/management/pages/blocklist/view/components/create_blocklist_form.tsx

```diff
@@ -0,0 +1,472 @@
+    }, [blocklistEntry.value]);
+
+    const osOptions: Array<EuiSuperSelectOption<OperatingSystem>> = useMemo(
+      () =>
+        [OperatingSystem.LINUX, OperatingSystem.MAC, OperatingSystem.WINDOWS].map((os) => ({
+          value: os,
+          inputDisplay: OS_TITLES[os],
+        })),
+      []
+    );
+
+    const fieldOptions: Array<EuiSuperSelectOption<ConditionEntryField>> = useMemo(() => {
+      const selectableFields: Array<EuiSuperSelectOption<ConditionEntryField>> = [
+        ConditionEntryField.HASH,
+        ConditionEntryField.PATH,
+      ].map((field) => ({
+        value: field,
+        inputDisplay: CONDITION_FIELD_TITLE[field],
+        dropdownDisplay: getDropdownDisplay(field),
+      }));
+      if (selectedOs === OperatingSystem.WINDOWS) {
+        selectableFields.push({
+          value: ConditionEntryField.SIGNER,
+          inputDisplay: CONDITION_FIELD_TITLE[ConditionEntryField.SIGNER],
+          dropdownDisplay: getDropdownDisplay(ConditionEntryField.SIGNER),
+        });
+      }
+
+      return selectableFields;
+    }, [selectedOs]);
+
+    const policyOptions = useMemo(() => {
+      return policiesRequest.data?.items || [];
+    }, [policiesRequest.data?.items]);
+
+    const validate = useCallback(
+      (field: ConditionEntryField, os: OperatingSystem, type: 'match_any', values: string[]) => {
+        if (!visited.value) {
+          return isValid;
+        }
+
+        const newWarnings: React.ReactNode[] = [];
+        const newErrors: React.ReactNode[] = [];
+
+        // error if empty
+        if (!values.length) {
+          newErrors.push(createValidationMessage(ERRORS.VALUE_REQUIRED));
+        }
+
+        // error if invalid hash
+        if (field === ConditionEntryField.HASH && values.some((value) => !isValidHash(value))) {
+          newErrors.push(createValidationMessage(ERRORS.INVALID_HASH));
+        }
+
+        const isInvalidPath = values.some((value) => !isPathValid({ os, field, type, value }));
+
+        // warn if invalid path
+        if (field !== ConditionEntryField.HASH && isInvalidPath) {
+          newWarnings.push(createValidationMessage(ERRORS.INVALID_PATH));
+        }
+
+        // warn if wildcard
+        if (
+          field !== ConditionEntryField.HASH &&
+          !isInvalidPath &&
+          values.some((value) => !hasSimpleExecutableName({ os, type, value }))
+        ) {
+          newWarnings.push(createValidationMessage(ERRORS.WILDCARD_PRESENT));
+        }
+
+        setWarnings({ ...warnings, value: newWarnings });
+        setErrors({ ...errors, value: newErrors });
+
+        return isValid && !newErrors.length;
+      },
+      [visited, warnings, errors, isValid]
+    );
+
+    const handleOnNameBlur = useCallback(() => {
+      if (!item.name.trim()) {
+        setErrors({ ...errors, name: [createValidationMessage(ERRORS.NAME_REQUIRED)] });
+      }
+      setVisited({ ...visited, name: true });
```

Comment:
If you want to use the prev visited state and not having it in the dependences array (avoid unnecessary renders of this useCallback) you can change this by:

```typescript
    const handleOnNameBlur = useCallback(() => {
      if (!item.name.trim()) {
        setErrors({ ...errors, name: [createValidationMessage(ERRORS.NAME_REQUIRED)] });
      }
      setVisited((prevVisited) => ({ ...prevVisited, name: true }));
   }, [item.name, errors]);
```